### ISSUE TITLE
mistake in Decoder wrt to non square images. 

### ISSUE THREAD

**Issue by saBasKouw** [jindongwang/Pytorch-CapsuleNet#4]
Hello Jin Dong Wang,

I noticed in the: 

`class Decoder(nn.Module):
    def __init__(self, input_width=28, input_height=28, input_channel=1):
        super(Decoder, self).__init__()
        self.input_width = input_width
        self.input_height = input_height
        self.input_channel = input_channel
        self.reconstraction_layers = nn.Sequential(
            nn.Linear(16 * LABEL_SIZE, 512),
            nn.ReLU(inplace=True),
            nn.Linear(512, 1024),
            nn.ReLU(inplace=True),
            nn.Linear(1024, self.input_height * self.input_height * self.input_channel),
            nn.Sigmoid()
        )`

The  line: ` nn.Linear(1024, self.input_height * self.input_height * self.input_channel), `

Should be: `nn.Linear(1024, self.input_height * self.input_width * self.input_channel), `

Do you agree?

**Comment by YuzhouLin** [comment]
Yes, it must be a typo mistake

**Comment by jindongwang** [comment]
Yes. A typo. Thanks!
